#!/usr/bin/env python
PACKAGE = "velo2cam_calibration"

from math import pi

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("x", double_t, 0, "x coord", 0, 0, 1)
gen.add("y", double_t, 0, "y coord", 0, 0, 1)
gen.add("z", double_t, 0, "z coord", 1, 0, 1)
gen.add("angle_threshold", double_t, 0, "Angle threshold for plane segmentation", 0.35, 0, pi/2)
gen.add("circle_radius", double_t, 0, "Radius of pattern's circles", 0.12, 0, 1)
gen.add("passthrough_radius_min", double_t, 0, "Min radius for passthrough", 1.0, 0, 10)
gen.add("passthrough_radius_max", double_t, 0, "Max radius for passthrough", 6.0, 0, 10)
gen.add("centroid_distance_min", double_t, 0, "Min distance to the centroid", 0.15, 0.0, 1.0)
gen.add("centroid_distance_max", double_t, 0, "Max distance to the centroid", 0.25, 0.0, 1.0)

exit(gen.generate(PACKAGE, "velo2cam_calibration", "Laser"))
